pytorch_lightning/trainer/connectors/logger_connector/result.py

```diff
@@ -356,7 +355,7 @@ class ResultCollection(dict):
     def __init__(self, training: bool, device: Optional[Union[str, torch.device]] = None) -> None:
         super().__init__()
         self.training = training
-        self._batch_size = torch.tensor(1, device=device)
+        self._current_batch = None
```

Comment:
I'm not a fan of this decision. Saving only the batch size value was done for memory and performance savings.

If each result object saves the current batch, then it's on more reference that we need to clean-up

Why can't the loop do it? I don't want the loop batch logic to leak into the ResultCollection

Reply:
this is cleaned inside `on_batch_end`. we somehow need to access the current batch inside `LightningModule.log`, can't find an alternative way for it. any suggestions?

Reply:
@carmocca We moved the logic within the ResultCollection as the other alternative was to keep this logic within the LightningModule.log method. This would make Trainer / LightningModule separation more complex.